base/dates/types.jl

```diff
@@ -19,10 +32,38 @@ for T in (:Hour,:Minute,:Second,:Millisecond)
     end
 end
 
+"""
+    Year(v)
+    Month(v)
+    Week(v)
+    Day(v)
+    Hour(v)
+    Minute(v)
+    Second(v)
+    Millisecond(v)
+
+Construct a `Period` type with the given `v` value. Input must be losslessly convertible
+to an `Int64`.
+"""
+Period(v)
+
 # Instant types represent different monotonically increasing timelines
```

Comment:
Done in https://github.com/JuliaLang/julia/pull/17417/commits/6535f7164419ad2a31dc6b05b4aea06de025d20f.